lib/pci.c

```diff
@@ -330,15 +321,50 @@ pci_nonstd_access(vfu_ctx_t *vfu_ctx, char *buf, size_t count,
  *
  * @cb is set to the callback to use for accessing the segment.
  */
-static size_t
+static ssize_t
 pci_config_space_next_segment(vfu_ctx_t *ctx, size_t count, loff_t offset,
-                              vfu_region_access_cb_t **cb)
+                              bool is_write, vfu_region_access_cb_t **cb)
 {
     struct pci_cap *cap;
+    static const size_t off2sz[] = {
```

Comment:
could you move this out to a utility function please?
Also, instead of hard-coding, you can do something like:

     vfu_pci_hdr_t hdr;                                                           
                                                                                 
     static const size_t off2sz[] = {                                             
       [offsetof(vfu_pci_hdr_t, bist)] = sizeof(hdr.bist)  
...                      
     };                                                                           

although you'd still need to account for the BAR array a bit more manually

Reply:
> could you move this out to a utility function please?

Sure.

> Also, instead of hard-coding, you can do something like:

OK.